sstables/sstables.hh

```diff
@@ -730,10 +732,18 @@ public:
     //
     // When created with `integrity_check::yes`, the integrity mechanisms
     // of the underlying data streams will be enabled.
+    //
+    // The `error_handler` parameter allows to customize the error handling
+    // logic when a checksum or digest mismatch is detected on an
+    // integrity-checked stream with no compression. The parameter is ignored
+    // if integrity checking is disabled or the SSTable is compressed.
     using raw_stream = bool_class<class raw_stream_tag>;
     input_stream<char> data_stream(uint64_t pos, size_t len,
             reader_permit permit, tracing::trace_state_ptr trace_state, lw_shared_ptr<file_input_stream_history> history,
-            raw_stream raw = raw_stream::no, integrity_check integrity = integrity_check::no);
+            raw_stream raw = raw_stream::no, integrity_check integrity = integrity_check::no,
+            std::optional<integrity_error_handler_t> error_handler = [] (sstring msg) {
+                throw sstables::malformed_sstable_exception(msg);
+            });
```

Comment:
The optional combined with the default parameter doesn't make sense. Either keep the optional and default to `std::nullopt`, or keep the default but make this a non-optional.